src/libraries/System.Memory/src/System/Buffers/Text/Base64Decoder.cs

```diff
@@ -644,6 +655,133 @@ private static unsafe void Ssse3Decode(ref byte* srcBytes, ref byte* destBytes,
+        {
+            // Implement an 8 way table lookup.
+            // This could be reduced by using two NEON TBX4 instructions.
+
+            Debug.Assert(AdvSimd.Arm64.IsSupported && BitConverter.IsLittleEndian);
+
+            Vector128<byte> dest = defaults;
+            Vector128<byte> indicies_sub = indicies;
+
+            dest = AdvSimd.Arm64.VectorTableLookupExtension(dest, table0, indicies_sub);
+            indicies_sub = AdvSimd.Subtract(indicies_sub, offset);
+            dest = AdvSimd.Arm64.VectorTableLookupExtension(dest, table1, indicies_sub);
+            indicies_sub = AdvSimd.Subtract(indicies_sub, offset);
+            dest = AdvSimd.Arm64.VectorTableLookupExtension(dest, table2, indicies_sub);
+            indicies_sub = AdvSimd.Subtract(indicies_sub, offset);
+            dest = AdvSimd.Arm64.VectorTableLookupExtension(dest, table3, indicies_sub);
+            indicies_sub = AdvSimd.Subtract(indicies_sub, offset);
+            dest = AdvSimd.Arm64.VectorTableLookupExtension(dest, table4, indicies_sub);
+            indicies_sub = AdvSimd.Subtract(indicies_sub, offset);
+            dest = AdvSimd.Arm64.VectorTableLookupExtension(dest, table5, indicies_sub);
+            indicies_sub = AdvSimd.Subtract(indicies_sub, offset);
+            dest = AdvSimd.Arm64.VectorTableLookupExtension(dest, table6, indicies_sub);
+            indicies_sub = AdvSimd.Subtract(indicies_sub, offset);
+            dest = AdvSimd.Arm64.VectorTableLookupExtension(dest, table7, indicies_sub);
+
+            return dest;
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static unsafe Vector128<byte> AdvSimdTbx3Byte(Vector128<byte> defaults, Vector128<byte> table0, Vector128<byte> table1, Vector128<byte> table2, Vector128<byte> indicies, Vector128<byte> offset)
+        {
+            // Implement a 3 way table lookup.
+
+            Debug.Assert(AdvSimd.Arm64.IsSupported && BitConverter.IsLittleEndian);
+
+            Vector128<byte> dest = defaults;
+            Vector128<byte> indicies_sub = indicies;
+
+            dest = AdvSimd.Arm64.VectorTableLookupExtension(dest, table0, indicies_sub);
+            indicies_sub = AdvSimd.Subtract(indicies_sub, offset);
+            dest = AdvSimd.Arm64.VectorTableLookupExtension(dest, table1, indicies_sub);
+            indicies_sub = AdvSimd.Subtract(indicies_sub, offset);
+            dest = AdvSimd.Arm64.VectorTableLookupExtension(dest, table2, indicies_sub);
+
+            return dest;
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static unsafe void AdvSimdDecode(ref byte* srcBytes, ref byte* destBytes, byte* srcEnd, int sourceLength, int destLength, byte* srcStart, byte* destStart)
```

Comment:
Just curious - does AggressiveInlining here show benefits in the benchmarks?

Reply:
If I remove this one it doesn't make a difference.
If I remove the two around AdvSimdTbx3Byte and AdvSimdTbx8Byte it gets 2x worse.

Reply:
those definitely make sense while this one is a bit questionable